### DISCUSSION TITLE
Fixture overlap

### DISCUSSION THREAD

**Discussion by Hooli486** [EnviralDesign/GeoPix#36] [Q&A]
I have no idea how GitHub forums normally work, so I have no idea if people rather want me to start a new discussion every time or if i should keep asking new questions inside older discussions which i started, but I'm just going to assume that this is fine haha :)

So, as said before, I'm a noob exploring the program and it's possibilities! 

I have the following problem, the two fixtures i created are overlapping. 
![afbeelding](https://user-images.githubusercontent.com/87446344/126380894-9c33cb7b-ee2a-4811-b37b-aad2e8505065.png)

However, I do not understand why they are overlapping. 
![afbeelding](https://user-images.githubusercontent.com/87446344/126381209-27258ccc-8fcc-4c52-99ea-9fe92f1f87f8.png)
![afbeelding](https://user-images.githubusercontent.com/87446344/126381222-2b95015f-8dea-4d91-8252-46aafd9e9df5.png)

As you can see, I made two 4x4 grid fixtures, with a pixel scale of 13. But, as the image shows, the grids don't overlap. Futhermore, you can see that one projector, with the size of one grid, also generates a response in the left grid which is not covered by the projector?

I'm not sure what I am doing wrong here, so if anybody quickly sees what i'm doing wrong and could help me on my way, that would be great :D 
In case more information is needed, let me know, and thanks in advance for helping!

**Comment by Hooli486** [comment]
![afbeelding](https://user-images.githubusercontent.com/87446344/126494569-8dafc2c1-c08c-41e9-9d50-b15320cb8dc1.png)

I tried the same in a simpler set-up, using two 10x10 grids, but the same happens. Error pop-ups that there is a fixture overlap, and the projector works in a weird way, of only lighting up pixels if the project is focussed on the top grid...

**Comment by Hooli486** [comment]
I just 'fixed' it, by putting my second fixture to device number two. I am not entirely sure if this is the normal way to fix this. But I gues it's kind of logical, as you say that each fixture is basically a different panel in real life, and thus needs a different device output. However, does this also mean you can only use maximum16 led panels in real life?

**Comment by EnviralDesign** [comment]
@Hooli486 hey again! This is a great place for questions or things you aren't sure are bugs or not. I can convert them to issues if it turns out they are. Make as many threads as you want if the topic is new or different, helps others find solutions to the same problems more easily.

---

So I'll start at the high level, you may already know some or all of this:

In GeoPix, you are ultimately sending DMX data to a pixel controller(s) that has some IP address (unless you're using sACN)
That pixel controller can have say 16 panels attached to different physical outputs, such as on the PixLite 16, and those outputs are configurable within the pixel controller. IE. how many pixels per output, starting address of each output, stuff like that.

So in GP if all of those 16 fixtures share the same device ID - it's telling GP that all of those fixtures exist within the same specified target IP address - aka the same pixel controller.

---

If you're seeing that message it's due to overlap not in physical space, but in address space. So if you have two fixtures that start at universe 1, channel 1, they are overlapping in address space.

You can view the Channels Tool, and see the flashing red areas where the fixtures are overlapping in address space which may be helpful.

You can also select 1 or more fixtures, and under the fixture dropdown in the viewport, run "Auto Offset Fixtures" which just makes up a non overlapping address for every fixture. Useful if you just want to fix the error and not worried about addressing at this time.

If you select your second fixture, and set it's universe to something different, does that error go away?

---

The weirdness you are seeing with projectors is due to the fact that both fixtures are attempting to write to the same place in the DMX data buffer. so what's happening is one fixture writes it's data, then the second fixture writes it's unique data to the same place, overwriting it.

The result on screen, is that things get scrambled.

**Reply by Hooli486** [reply]
Thank you very much for the elaborate answer, definitely learned from it! The auto offset fixture indeed fixed it, I did not think about checking the channels tool at all, but the adress space story makes sense :)
Also nice to understand the background of the projectors and why it went wrong, thanks for all the help!
